lib/matplotlib/tests/test_backends_interactive.py

```diff
@@ -186,3 +186,18 @@ def test_webagg():
     conn.close()
     proc.send_signal(signal.SIGINT)
     assert proc.wait(timeout=_test_timeout) == 0
+
+
+@pytest.mark.backend('TkAgg', skip_on_importerror=True)
+def test_never_update(monkeypatch):
+    import tkinter
+    monkeypatch.delattr(tkinter.Misc, 'update')
+    monkeypatch.delattr(tkinter.Misc, 'update_idletasks')
+
+    import matplotlib.pyplot as plt
+    plt.plot([1, 2, 3], [1, 3, 5])
+    plt.show(block=False)
+    fig = plt.gcf()
+    fig.canvas.toolbar.configure_subplots()
+    # skirt monkeypatch
```

Comment:
How about using `monkeypatch.context()` instead?

Reply:
Unfortunately that doesn't have quite the right behavior here. I need to run the event loop to trigger some callbacks that could (and formerly did) call `Misc.update()`.  I chose this instead of `plt.pause(1)` for efficiency and robustness. A marginally less magical way to do it might be 

```fig.canvas.get_tk_widget().tk.call('update')```

What do you think of that?